.github/workflows/lint.yml

```diff
@@ -1,44 +1,13 @@
-# https://docs.github.com/en/actions
-name: Lint
+name: lint
+
 
 on: [pull_request]
 
 jobs:
-  misspell:
-    name: Check spelling all files in commit misspell # https://github.com/client9/misspell
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check out
-        uses: actions/checkout@v3
-      - name: Install
-        run: wget -O - -q https://git.io/misspell | sh -s -- -b . # https://www.gnu.org/software/wget/
-      - name: misspell
-        run: git ls-files --empty-directory | xargs ./misspell -error
   pre-commit:
     name: Run pre-commit # https://pre-commit.com/
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4 # https://www.python.org/
-        with:
-          python-version: '3.x' # Version range or exact version of a Python version to use, using SemVer's version range syntax
-          architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
-      - name: Install dependencies # https://pip.pypa.io/en/stable/
-        run: |
-          python -m pip install --upgrade pip
-          pip install pre-commit
-      - name: Set PY
-        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
-      - uses: actions/cache@v3
-        with:
-          path: ~/.cache/pre-commit
-          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
-      - name: Run pre-commit
-        run: pre-commit run --all-files
-  trailing-whitespace:
-    name: Trailing whitespace # https://en.wikipedia.org/wiki/Grep
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Check for trailing whitespace
-        run: "! git grep -EIn $'[ \t]+$' | grep -v '\\- '"
+      - uses: actions/setup-python@v3
```

Comment:
why are we downgrading this?

Reply:
now we are using the action to run the pre-commit, instead of installing it and running the command on a custom action 
https://github.com/marketplace/actions/pre-commit

the action is stated to be deprecated for new features but for what we are needing here i think is enough to run the checks. WDYT?

Reply:
I think we should use the latest version of the action and pin it to an specific python version. That way we control which python version we are executing and can troubleshoot any problem

Reply:
yes, i reverted this... the change was made beacuse i changed to use the pre-commit action instead of the installation at the action, and at the example they are using v3

cc: @ifbyol